File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.11-pattern.js
function log(val) {
    process.stdout.write(String(val));
}
//

function pattern_11() {
    let row = 10;
    let column = 5;

    for (let y = 1; y < row; y++) {
        for (let x = 1; x < column + 1; x++) {
            if (y <= row / 2) {
                if (x <= y) {
                    log('*');
                } else {
                    log(' ');
                }
            } else {
                if (x <= row - y) {
                    log('*');
                } else {
                    log(' ');
                }
            }
        }
        console.log();
    }
}

pattern_11();

File: python/0.0-Basics/0.3-build-up-logical-thinking/0.3.11-pattern.py
from utils.clear import clear


def pattern_11():
    row = 10
    column = 5

    for i in range(1, row):
        for j in range(1, column + 1):
            if i <= row / 2:
                if j <= i:
                    print("*", end="")
                elif j > i:
                    print(" ", end="")

            elif i > row / 2:
                key = round(row  - i)
                if j <= key:
                    print("*", end="")
                elif j > key:
                    print(" ", end="")

        print()


clear()
pattern_11()
